### ISSUE TITLE
ggplot to ggplotly doesn't work with customized 'fill' values in R

### ISSUE THREAD

**Issue by speedreader** [ropensci/plotly#421]
Trying to use plotly and ggplot to make a geom_bar with a custom colored bar. 

    set.seed(8)
    y <- rpois(1000, 2.9)
    table(y)

    holdThis <- as.data.frame(table(y))
    holdThis$cumTotal <- cumsum(holdThis$Freq)
    holdThis$percentage <- (as.numeric(holdThis$Freq)/10000) * 100
    holdThis$y <- as.numeric(as.character(holdThis$y))

    # In practice this will be generated dynamically, but for here I'll just fill them manually.
    fillCol <- c('blue', 'blue','blue','blue','blue','blue','red','blue','blue','blue')

    p <- ggplot(holdThis, aes(x=y, y=Freq)) + geom_bar(stat='identity', fill= fillCol, color="black", size = .25, alpha = .7) + scale_x_discrete(breaks = y)

    ggplotly(p)
 
ggplot produces this (works):

<img width="868" alt="screen shot 2016-01-23 at 3 14 04 pm" src="https://cloud.githubusercontent.com/assets/16739809/12532892/e5675cb4-c1e4-11e5-8f17-22a86223f5de.png">

ggplotly(p) produces this message:
Warning message:
In if (!grepl(pat, x)) return(x) :
  the condition has length > 1 and only the first element will be used

And this:
<img width="1223" alt="screen shot 2016-01-23 at 3 14 26 pm" src="https://cloud.githubusercontent.com/assets/16739809/12532893/e97e1824-c1e4-11e5-8d13-caca97f6a240.png">

**Comment by cpsievert** [comment]
Bug confirmed, thanks.

**Comment by yonicd** [comment]
I have same issue with geom_boxplots when the fill variable has a different number of levels than the x axis variable

```r
set.seed(1234)
dat <- data.frame(cond = factor(rep(c("A","B"), each=200)), 
                            cond2 = factor(rep(c("C","D","E","F"), each=100)),
                             rating = c(rnorm(200),rnorm(200, mean=.8))
)

dat2 <- data.frame(cond = factor(rep(c("A","B"), each=200)),
                  cond2 = factor(rep(c("C","D"), each=200))
                  rating = c(rnorm(200),rnorm(200, mean=.8))
                  )

p1=ggplot(dat, aes(x=cond, y=rating, fill=cond2)) + geom_boxplot()
p2=ggplot(dat2, aes(x=cond, y=rating, fill=cond2)) + geom_boxplot()

ggplotly(p1)
Warning messages:
1: In if (!grepl(pat, x)) return(x) :
  the condition has length > 1 and only the first element will be used
2: In if (!grepl(pat, x)) return(x) :
  the condition has length > 1 and only the first element will be used

ggplotly(p2)
```

![image](https://cloud.githubusercontent.com/assets/1755487/13298570/f60dcd9a-db3f-11e5-97ee-1a2743c6f947.png)

![image](https://cloud.githubusercontent.com/assets/1755487/13298664/61cdc2ce-db40-11e5-9c88-085685b50c06.png)

**Comment by yonicd** [comment]
there bug looks to be in line 630 of the gg2list function
you need to wrap the logical with any to make it work for multiple comparisons

```r
  rm_alpha <- function(x) {
    if (length(x) == 0) 
      return(x)
    pat <- "^rgba\\("
    if (any(!grepl(pat, x)))  # <-------
      return(x)
    sub(",\\s*[0]?[.]?[0-9]+\\)$", ")", sub(pat, "rgb(", 
                                            x))
  }
```

the fixed copy of the function is [here](https://github.com/yonicd/CIMDO/blob/master/gg2list.r)

**Comment by cpsievert** [comment]
This should be fixed in f1900cc3cd464408a60ffadfe3e95460747430e0. Let us know if you still have problems.

**Comment by yonicd** [comment]
much better but there is still a bug. attached is sample data.

running the code below brings up the correct graph but when toggling off any of the fill factors (eg yr==2006) interactively it causes the last filled (right most) boxplot to increase in width.

```r
library(plotly)
data=read.csv("https://raw.githubusercontent.com/yonicd/CIMDO/master/temp/example.csv",header = T,stringsAsFactors = F)
ggplot(data,aes(x=cut0,y=y3,fill=yr))+geom_boxplot()
ggplotly()
```
